src/main/java/com/redhat/devtools/lsp4ij/settings/jsonSchema/MyLightVirtualFile.java

```diff
@@ -0,0 +1,187 @@
+// Copyright 2000-2023 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package com.redhat.devtools.lsp4ij.settings.jsonSchema;
+
+import com.intellij.lang.Language;
+import com.intellij.openapi.fileTypes.CharsetUtil;
+import com.intellij.openapi.fileTypes.FileType;
+import com.intellij.openapi.fileTypes.FileTypeRegistry;
+import com.intellij.openapi.util.NlsSafe;
+import com.intellij.openapi.util.io.FileTooBigException;
+import com.intellij.openapi.util.io.FileUtilRt;
+import com.intellij.openapi.vfs.VfsUtilCore;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.openapi.vfs.limits.FileSizeLimit;
+import com.intellij.testFramework.LightVirtualFileBase;
+import com.intellij.util.ArrayUtil;
+import com.intellij.util.LocalTimeCounter;
+import com.intellij.util.ThreeState;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.*;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * In-memory implementation of {@link VirtualFile}.
+ */
+public class MyLightVirtualFile extends LightVirtualFileBase {

```

Comment:
Should this be called `LSPJsonSchemaLightVirtualFile` or similar? Also, can/should its visibility be package-protected instead of public?

Reply:
fixed